src/pkgcheck/checks/python.py

```diff
@@ -725,6 +744,101 @@ def feed(self, pkg):
                     yield PythonGHDistfileSuffix(f.filename, uri, pkg=pkg)
                     break
 
+    def simplify_pn_pv(self, pn: str, pv: str, pkg, allow_none: bool
+                       ) -> typing.Tuple[str, str]:
+        if pv == pkg.version:
+            pv = None if allow_none else '"${PV}"'
+        # check for common PV transforms
+        elif pv == pkg.version.replace("_p", ".post", 1):
+            pv = '"${PV/_p/.post}"'
+        elif pv == pkg.version.replace("_alpha", "a", 1):
+            pv = '"${PV/_alpha/a}"'
+        elif pv == pkg.version.replace("_beta", "b", 1):
+            pv = '"${PV/_beta/b}"'
+
+        if pn == pkg.package:
+            pn = None if pv is None else '"${PV}"'
+        # check for common PN transforms that conform to naming policy
+        elif pn == pkg.package.replace("-", ".", 1):
+            pn = '"${PN/-/.}"'
+        elif pn == pkg.package.replace("-", "."):
+            pn = '"${PN//-/.}"'
+        elif pn == pkg.package.replace("-", "_", 1):
+            pn = '"${PN/-/_}"'
+        elif pn == pkg.package.replace("-", "_"):
+            pn = '"${PN//-/_}"'
+        # .title() is not exactly the same as ^
+        elif pn == f"{pkg.package[:1].upper()}{pkg.package[1:]}":
+            pn = '"${PN^}"'
+
+        return pn, pv
+
+    def check_pypi_mirror(self, pkg, fetchables):
+        # consider only packages that don't inherit pypi.eclass already
+        if "pypi" in pkg.inherited:
+            return
+
+        uris = [uri for f in fetchables for uri in f.uri]
+        # check if we have any mirror://pypi URLs in the first place
+        for uri in uris:
+            if uri.startswith(PYPI_URI_PREFIX):
+                break
+        else:
+            return
+
+        # if there's exactly one URI, perhaps inheriting the eclass will suffice
+        if len(uris) == 1:
+            expected_fn = f"{pkg.package}-{pkg.version}.tar.gz"
+            expected = f"{PYPI_URI_PREFIX}source/{pkg.package[0]}/{pkg.package}/{expected_fn}"
+            if uris[0] == expected and next(iter(fetchables)).filename == expected_fn:
+                yield PythonInlinePyPIURI(uris[0], pkg=pkg)
+                return
+
+        # otherwise, yield result for every URL, with suggested replacement
+        for f in fetchables:
+            for uri in f.uri:
+                source_match = PYPI_SDIST_URI_RE.match(uri)
+                if source_match is not None:
+                    pn, pv, suffix = source_match.groups()
+
+                    if suffix == ".tar.gz":
+                        suffix = None
+                    pn, pv = self.simplify_pn_pv(pn, pv, pkg, suffix is None)
+
+                    args = [arg for arg in ("pypi_sdist_url", pn, pv, suffix) if arg is not None]
```

Comment:
I don't know if this considered better? But consider:

`args = tuple(filter(None, ("pypi_sdist_url", pn, pv, suffix)))`

(When filter get's `None` as function, it filters all those that `bool(` return True, in this case as string it will drop `None` and `""`, and `""` is impossible from the function, so looks equivalent).

Reply:
Well, it ends up shorter, so I'm happy with it ;-).